### PR TITLE
Rename extendType to extendedType and deprecate extendType

### PR DESCRIPTION
**Pull request by soham-bentley**
Fixes https://github.com/iTwin/itwinjs-backlog/issues/1291

### CODE REVIEW

File: core/common/src/ConcurrentQuery.ts [hunk 1]
```diff
@@ -61,8 +61,12 @@ export interface QueryPropertyMetaData {
   jsonName: string;
   /** The name is the property's alias if the property is a generated one, otherwise, it is the name of the property. */
   name: string;
-  /** If this property is a PrimitiveECProperty, extend type is the extended type name of this property, if it is not defined locally will be inherited from base property if one exists, otherwise extended type is set to an empty string. */
+  /** If this property is a PrimitiveECProperty, extend type is the extended type name of this property, if it is not defined locally will be inherited from base property if one exists, otherwise extend type is set to an empty string.
+   * @deprecated in 4.10 Use extendedType instead
```
Comment (soham-bentley) [inline core/common/src/ConcurrentQuery.ts:65]: I have written @deprecated in 4.10    Is the version number okay? Or should I write something else?
@khanaffan

Comment (aruniverse) [reply core/common/src/ConcurrentQuery.ts:65]: 4.10 is already released. It'll have to go in 4.11

**Review by khanaffan — APPROVED** [review]

File: core/common/src/ConcurrentQuery.ts (hunk 1 above)
Comment (soham-bentley) [reply core/common/src/ConcurrentQuery.ts]: Changed the version to 4.11

File: common/changes/@itwin/core-backend/Soham-extendedTypeFixConcurrentQuery_2024-11-13-14-06.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/core-backend",
+      "comment": "Added test case to document the behaviour of properties extendType and extendedType of QueryPropertyMetaData",
```
Comment (pmconne) [inline common/changes/@itwin/core-backend/Soham-extendedTypeFixConcurrentQuery_2024-11-13-14-06.json:5]:
```suggestion
      "comment": "",
```

End users don't care about tests.

**Review by pmconne — APPROVED** [review]

**Comment by aruniverse** [comment]
@mergifyio backport release/4.11.x

**Pull request merged**